src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<? extends Machine> bulldozers = new BulldozerProducer().get();
+            return new ArrayList<>(bulldozers);
+        } else if (type == Excavator.class) {
+            List<? extends Machine> excavators = new ExcavatorProducer().get();
+            return new ArrayList<>(excavators);
+        }
+        List<? extends Machine> trucks = new TruckProducer().get();
+        return new ArrayList<>(trucks);
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        int size = machines.size();
+        for (int i = 0; i < size; i++) {
```

Comment:
Don't create a new variable. Immediately pass the machines.size() to the loop

Reply:
done